filament/src/components/RenderableManager.cpp

```diff
@@ -596,14 +596,22 @@ static void updateMorphWeights(FEngine& engine, backend::Handle<backend::HwBuffe
     driver.updateBufferObject(handle, { out, size }, 0);
 }
 
-void FRenderableManager::setMorphWeights(Instance instance, float const* weights, size_t count) noexcept {
+void FRenderableManager::setMorphTargetCount(Instance instance, size_t count) noexcept {
     if (instance) {
+        ASSERT_PRECONDITION(count < CONFIG_MAX_MORPH_TARGET_COUNT,
+                "Only %d morph targets are supported (count=%d)", CONFIG_MAX_MORPH_TARGET_COUNT, count);
+
         MorphWeights& morphWeights = mManager[instance].morphWeights;
         morphWeights.count = count;
+    }
+}
 
+void FRenderableManager::setMorphWeights(Instance instance, float const* weights, size_t count) noexcept {
+    if (instance) {
         ASSERT_PRECONDITION(count < CONFIG_MAX_MORPH_TARGET_COUNT,
```

Comment:
same here, can't be `noexcept` if `ASSERT_*` is used.

Reply:
Thanks it's updated.